src/cli/config/config_test.go

```diff
@@ -223,6 +243,104 @@ func (suite *ConfigSuite) TestMustMatchConfig() {
+		passphrase             = "passphrase-test"
+		azureClientID          = "azure-client-id-test"
+		azureSecret            = "azure-secret-test"
+		prefix                 = "prefix-test"
+		disableTLS             = "true"
+		disableTLSVerification = "true"
+	)
+
+	// Generate test config file
+	testConfigData := fmt.Sprintf(configFileTemplate, b, tID, accKey, secret, token,
+		passphrase, azureClientID, azureSecret,
+		disableTLS, disableTLSVerification)
+
+	testConfigFilePath := filepath.Join(t.TempDir(), "corso.toml")
+	err := os.WriteFile(testConfigFilePath, []byte(testConfigData), 0o700)
+	require.NoError(t, err, clues.ToCore(err))
+
+	// Configure viper to read test config file
+	vpr.SetConfigFile(testConfigFilePath)
+
+	// Read and validate config
+	err = vpr.ReadInConfig()
+	require.NoError(t, err, "reading repo config", clues.ToCore(err))
+
+	overrides := map[string]string{}
+	flags.AzureClientTenantFV = "6f34ac30-8196-469b-bf8f-d83deadbbbba"
+	flags.AzureClientIDFV = "azure-id-flag-value"
+	flags.AzureClientSecretFV = "azure-secret-flag-value"
+
+	flags.AWSAccessKeyFV = "aws-access-key"
+	flags.AWSSecretAccessKeyFV = "aws-access-secret-flag-value"
+	flags.AWSSessionTokenFV = "aws-access-session-flag-value"
+
+	overrides[storage.Bucket] = "flag-bucket"
+	overrides[storage.Endpoint] = "flag-endpoint"
+	overrides[storage.Prefix] = "flag-prefix"
+	overrides[storage.DoNotUseTLS] = "true"
+	overrides[storage.DoNotVerifyTLS] = "true"
+	overrides[credentials.AWSAccessKeyID] = flags.AWSAccessKeyFV
+	overrides[credentials.AWSSecretAccessKey] = flags.AWSSecretAccessKeyFV
+	overrides[credentials.AWSSessionToken] = flags.AWSSessionTokenFV
+
+	flags.CorsoPassphraseFV = "passphrase-flags"
+
+	repoDetails, err := getStorageAndAccountWithViper(
+		vpr,
+		true,
+		false,
+		overrides,
+	)
+
+	m365Config, _ := repoDetails.Account.M365Config()
+	s3Cfg, _ := repoDetails.Storage.S3Config()
+	commonConfig, _ := repoDetails.Storage.CommonConfig()
+	pass := commonConfig.Corso.CorsoPassphrase
+
+	require.NoError(t, err, "reading repo config", clues.ToCore(err))
+
+	assert.Equal(t, flags.AWSAccessKeyFV, s3Cfg.AWS.AccessKey)
+	assert.Equal(t, flags.AWSSecretAccessKeyFV, s3Cfg.AWS.SecretKey)
+	assert.Equal(t, flags.AWSSessionTokenFV, s3Cfg.AWS.SessionToken)
+
+	assert.Equal(t, overrides[storage.Bucket], s3Cfg.Bucket)
+	assert.Equal(t, overrides[storage.Endpoint], s3Cfg.Endpoint)
+	assert.Equal(t, overrides[storage.Prefix], s3Cfg.Prefix)
+	assert.Equal(t, str.ParseBool(overrides[storage.DoNotUseTLS]), s3Cfg.DoNotUseTLS)
+	assert.Equal(t, str.ParseBool(overrides[storage.DoNotVerifyTLS]), s3Cfg.DoNotVerifyTLS)
+
+	assert.Equal(t, flags.AzureClientIDFV, m365Config.AzureClientID)
+	assert.Equal(t, flags.AzureClientSecretFV, m365Config.AzureClientSecret)
+	assert.Equal(t, flags.AzureClientTenantFV, m365Config.AzureTenantID)
+
+	assert.Equal(t, flags.CorsoPassphraseFV, pass)
+
+	// reset values
+	flags.AzureClientTenantFV = ""
```

Comment:
this makes me a little uneasy, because it can be easy to cross-contaminate other tests unintentionally.  In the long, long term we'll probably not want to use globals for flag values.  But that's out of scope of this change.

For this test, can you please move these resets into a deferred func on line 290?  That'll ensure that this gets changed even in the event of an early failure  Or maybe better than that, the tester library has an "after every test" hook that you can put these into.

Reply:
Yes. both these things make sense. Global flag values is not that great idea. Meanwhile I will move those statements to defer.